### ISSUE TITLE
Stabilize URL format

### ISSUE THREAD

**Issue by gwhitney** [numberscope/frontscope#439]
Before ui2->main and deploying, a "final" URL format should be selected, and the code modified to support that and only that throughout (this will allow pruning some code to handle e.g. base64-encoded URLs).

**Comment by gwhitney** [comment]
I just realized @katestange and @Vectornaut that the _filename_ of a visualizer (maybe a sequence, too) is the _value_ of the viz URL query parameter to select that visualizer. Is that the convention we want? Meaning that changing the filename of a source file will necessitate semantic version increase? Maybe so, but I just thought we should contemplate that decision before setting it in stone, so to speak.

And if that is our decision, may I request we rename Histogram to FactorHistogram? I'd like to leave the "Histogram" identifier free for a future general-purpose Histogram visualizer.

**Comment by katestange** [comment]
+1 on the renaming Histogram to FactorHistogram.  I also don't feel completely happy with FactorFence... can anyone think of a better more descriptive name?  The name came from the picket fence appearance, but it doesn't really explain what it is.

**Comment by gwhitney** [comment]
> happy with FactorFence...

As counterpoint, I like the name, I think it's charming. But just to brainstorm alternatives (not that I think they're necessarily better): PrimeLog, FactorLog, FactorLogs (I sorta like the mild pun between a log as a record and a log as a post); TotemPrimes (maybe runs afoul of cultural sensitivity, though?); FactorStacks (nice internal rhyme); PrimePosts/LogPosts/FactorPosts; LogBars/PrimeBars/FactorBars; ... ?

**Comment by Vectornaut** [comment]
I agree that FactorFence is charming and memorable! I also like FactorStacks, which is very descriptive.

Requiring that the filename of the visualizer match the desired URL parameter name doesn't sound like too much of a burden to me. The only potential pitfall I can think of is that maybe we'll want the URL parameter to contain a character that's not allowed in filenames... but that feels far-fetched to me.

**Comment by gwhitney** [comment]
OK, this is next up, so @katestange it's time to decide if we're sticking with FactorFence as the name.

**Comment by katestange** [comment]
Yes, I think I like FactorStacks also!  Can we go for that one?  It's a bit more descriptive.

**Comment by gwhitney** [comment]
> Can we go for that one?

It's your visualizer (and generally speaking, the Numberscope buck stops with you ;-) so sure. Whoever gets to it first should rename FactorFence -> FactorStacks.

**Comment by gwhitney** [comment]
Some additional detail on the plans here: we will put the git commit in the URL as a backup but mostly ignore it on load; we will put a URL version number in the URL, and register URL update rules, which if they fail, will produce useful error messages for the visitor. The URL version number must be bumped for any code change that invalidates any previously working URL.
